src/Types/ReceiveName.cs

```diff
@@ -34,6 +34,18 @@ public static bool TryParse(string name, out (ReceiveName? ReceiveName, Validati
         return validate;
     }
 
+    /// <summary>
+    /// Get the contract name part of <see cref="Receive"/>.
+    /// </summary>
+    /// <returns>Contract name</returns>
+    public string GetContractName() => this.Receive[..this.Receive.IndexOf('.')];
```

Comment:
I think this should return a `ContractName` instead of a string.

Reply:
Unfortunately we already have a type called `ContractName`. It follows the same naming conventions as in base library
https://github.com/Concordium/concordium-base/blob/a3a16407c82ad49de404b3968ba8507f86314681/smart-contracts/contracts-common/concordium-contracts-common/src/types.rs#L1180
This type does however contain both prefix `init_` and the contract name.

Looking at it now aren't the naming wrong in both SDKs? Contract name should only be the part without the `init_` prefix as far as I understand.

I see two options
* Make a new type called something like `ContractNamePart`
* Change name of old type from `ContractName` to `InitName` and make a major bump

Reply:
> Looking at it now aren't the naming wrong in both SDKs? Contract name should only be the part without the init_ prefix as far as I understand.

Yes, that is how it is done in the JS SDK, but not necessarily the best solution.

> I see two options
> 
> - Make a new type called something like `ContractNamePart`
> - Change name of old type from `ContractName to `InitName` and make a major bump

Since `ContractName` and `InitName` are 1 to 1, and the only difference is how it is represented as a string, I think we should try to make the current `ContractName` represent both without breaking changes. It will simplify the API since there won't be two types for essentially the same information.

Reply:
Resolved through slack. We introduce a new type, since for a lot of use cases the user only have the contract name part. Ex. when working with contract- updated and interrupted events.